website/content/en/preview/concepts/threat-model.md

```diff
@@ -0,0 +1,87 @@
+* [clusterrole-core.yaml](https://github.com/aws/karpenter/blob{{< githubRelRef >}}charts/karpenter/templates/clusterrole-core.yaml)
+* [clusterrole.yaml](https://github.com/aws/karpenter/blob{{< githubRelRef >}}charts/karpenter/templates/clusterrole.yaml)
+* [rolebinding.yaml](https://github.com/aws/karpenter/blob{{< githubRelRef >}}charts/karpenter/templates/rolebinding.yaml)
+* [role.yaml](https://github.com/aws/karpenter/blob{{< githubRelRef >}}charts/karpenter/templates/role.yaml)
+
+## Assumptions
+
+| Category	| Assumption	| Comment	|
+| ---	| ---	| ---	|
+| Generic	| The Karpenter pod is operated on a node in the cluster, and uses a Service Account for authentication to the Kubernetes API	| Cluster Operators may want to isolate the node running the Karpenter pod to a system-pool of nodes to mitigate the possibility of container breakout with Karpenter’s permissions. 	|
+| Generic	| Cluster Developer does not have any Kubernetes permissions to manage Karpenter running in the cluster (The deployment, pods, clusterrole, etc)	|	|
+| Generic	| Restrictions on the fields of pods a Cluster Developer can create are out of scope. 	| Cluster Operators can use policy frameworks to enforce restrictions on Pod capabilities	|
+| Generic	| No sensitive data is included in non-Secret resources in the Kubernetes API. The Karpenter controller has the ability to list all pods, nodes, deployments, and many other pod-controller and storage resource types.	| Karpenter does not have permission to list/watch cluster-wide ConfigMaps or Secrets	|
+| Generic	| Karpenter has permissions to create, modify, and delete nodes from the cluster, and evict any pod. 	| Cluster Operators running applications with varying security profiles in the same cluster may want to configure dedicated nodes and scheduling rules for Karpenter to mitigate potential container escapes from other containers	|
+| AWS-Specific	| The Karpenter IAM policy is encoded in the GitHub repo. Any additional permissions possibly granted to that role by the administrator are out of scope	|	|
+| AWS-Specific	| The Karpenter pod uses IRSA for AWS credentials 	| Setup of IRSA is out of scope for this document 	|
+
+## Generic Threats and Mitigations
+
+### Threat: Cluster Developer can influence creation of an arbitrary number of nodes
+
+**Background**: Karpenter creates new instances based on the count of pending pods.
+
+**Threat**: A Cluster Developer attempts to have Karpenter create more instances than intended by creating a large number of pods or by using anti-affinity to schedule one pod per node.
+
+**Mitigation**: In addition to [Kubernetes resource limits](https://kubernetes.io/docs/concepts/policy/resource-quotas/#object-count-quota), Cluster Operators can [configure limits on a Provisioner](https://karpenter.sh/preview/concepts/provisioners/#speclimitsresources) to limit the total amount of memory, CPU, or other resources provisioned across all nodes.
+
+## AWS-Specific Threats
```

Comment:
It's mostly for future proofing. As new providers get written, they can naturally get added here without any rewrites

Reply:
Got it. I could go either way. I was mostly thinking about it because we've stopped drawing the line in the documentation between "AWS-specific" things and things that we might think of as vendor-neutral, mainly because it's yet to become clear how this might be delineated in the long-term.